pandas/tests/indexes/test_base.py

```diff
@@ -896,6 +896,33 @@ def test_difference(self):
         assert len(result) == 0
         assert result.name == first.name
 
+    def test_difference_sorting_false(self):
```

Comment:
* Reference issue number
* Will need test cases that cause your code to issue the warnings you added

Reply:
@gfyoung any changes required?

Reply:
@Giftlin Some cases will be unsorted even if `sort=True`.
Like this.
https://github.com/pandas-dev/pandas/pull/17878/files#diff-248241e14f081931cc805f9d0e137a5fR2211